native/src/main/java/io/ballerina/persist/nodegenerator/SyntaxTreeGenerator.java

```diff
@@ -0,0 +1,203 @@
+import static io.ballerina.persist.PersistToolsConstants.DEFAULT_USER;
+import static io.ballerina.persist.PersistToolsConstants.KEY_DATABASE;
+import static io.ballerina.persist.PersistToolsConstants.KEY_HOST;
+import static io.ballerina.persist.PersistToolsConstants.KEY_PASSWORD;
+import static io.ballerina.persist.PersistToolsConstants.KEY_PORT;
+import static io.ballerina.persist.PersistToolsConstants.KEY_USER;
+
+
+/**
+ * Class to create syntax tree for Config.toml.
+ */
+public class SyntaxTreeGenerator {
+    private static final PrintStream outStream = System.out;
+    private static final String[] nodeMap = {KEY_HOST, KEY_PORT, KEY_USER, KEY_DATABASE, KEY_PASSWORD};
+    private static final String[] defaultValues = {DEFAULT_HOST, DEFAULT_PORT, DEFAULT_USER, DEFAULT_DATABASE,
+            DEFAULT_DATABASE};
+
+    /**
+     * Method to create a new Config.toml file with database configurations.
+     */
+    public static SyntaxTree createToml(String name) {
+        NodeList moduleMembers = AbstractNodeFactory.createEmptyNodeList();
+        moduleMembers = moduleMembers.add(SampleNodeGenerator.createTable(name, null));
+        moduleMembers = populateNodeList(moduleMembers);
+        Token eofToken = AbstractNodeFactory.createIdentifierToken("");
+        DocumentNode documentNode = NodeFactory.createDocumentNode(moduleMembers, eofToken);
+        TextDocument textDocument = TextDocuments.from(documentNode.toSourceCode());
+        return SyntaxTree.from(textDocument);
+    }
+
+    /**
+     * Method to update the Config.toml with database configurations.
+     */
+    public static SyntaxTree updateToml(Path configPath, String name) throws IOException {
+
+        ArrayList<String> existingNodes = new ArrayList<>();
+        NodeList<DocumentMemberDeclarationNode> moduleMembers = AbstractNodeFactory.createEmptyNodeList();
+        Path fileNamePath = configPath.getFileName();
+        TextDocument configDocument = TextDocuments.from(Files.readString(configPath));
+        SyntaxTree syntaxTree = SyntaxTree.from(configDocument, fileNamePath.toString());
+        DocumentNode rootNote = syntaxTree.rootNode();
+        NodeList nodeList = rootNote.members();
+
+        for (Object member : nodeList) {
+            if (member instanceof KeyValueNode) {
+                moduleMembers = moduleMembers.add((DocumentMemberDeclarationNode) member);
+            } else if (member instanceof TableNode) {
+                TableNode node = (TableNode) member;
+                if (node.identifier().toString().trim().equals(name)) {
+                    NodeList subNodeList = node.fields();
+                    moduleMembers = moduleMembers.add(SampleNodeGenerator.createTable(name, null));
+                    for (Object subMember : subNodeList) {
+                        KeyValueNode subNode = (KeyValueNode) subMember;
+                        if (!isDatabaseConfigurationEntry(subNode.identifier())) {
```

Comment:
```suggestion
                        if (!isDatabaseConfigurationEntry(subMember.identifier())) {
```

Reply:
This casting is required to use `identifier()` method

Reply:
But the `subMember` is already of `KeyValueNode` type and being casted by `KeyValueNode` to create another `KeyValueNode` type object?

Reply:
Yes, but for iterating through the NodeList, each item has to be referred by Object type, so it has to be casted to `KeyValueNode` again to use `identifier()` without compilation errors

Reply:
In the loop can't we simply use `for (KeyValueNode subMember : subNodeList) {`